scripts/db_migrator.py

```diff
@@ -935,6 +935,27 @@ def migrate_dhcp_servers_to_dhcpv4_relay(self):
             except Exception as e:
                 log.log_error(f"Failed to migrate DHCP servers for {vlan_key}: {str(e)}")
 
+    def migrate_ipinip_tunnel_ecn_mode_mellanox(self):
+        """
+        Migrate IPINIP_TUNNEL ecn mode to copy_from_outer.
+
+        Prior to 202511, on Mellanox platform, the ecn mode of IPINIP tunnel was set to standard.
+        """
+
+        table_name = 'TUNNEL_DECAP_TABLE'
+        keys_to_migrate = [
+            'IPINIP_SUBNET',
+            'IPINIP_TUNNEL',
+            'IPINIP_SUBNET_V6',
+            'IPINIP_V6_TUNNEL'
+        ]
+
+        table = self.appDB.get_table(table_name)
+        for k, v in table.items():
+            if k in keys_to_migrate:
+                log.log_info(f"Migrating {k}, ecn mode: {v.get('ecn_mode')} -> copy_from_outer")
+                v['ecn_mode'] = 'copy_from_outer'
+                self.appDB.set_entry(table_name, k, v)
```

Comment:
setting the entry in the app db is enough? Will asic catch this?

Reply:
Yes, setting it into APPL_DB is enough, it will be configured in the _OrchDaemon::warmRestoreAndSyncUp()_

Also, ASIC db is flushed, so it's not relevant: https://github.com/sonic-net/sonic-utilities/blob/67232b63ac6486a7f320fd23b6e85245ad938b61/scripts/fast-reboot#L257

Regarding STATE db, I'm not sure I understand why it's relevant for this fix.

Reply:
@developfast, I've added a migration for the STATE DB. Regarding the ASIC DB, since it's flushed before migration (and will be filled after), there is nothing to migrate. Please correct me if I'm wrong.